### ISSUE TITLE
RFP - Winamp Playlist File Remover v2.5

### ISSUE THREAD

**Issue by EpsilonsQc** [chocolatey-community/chocolatey-package-requests#845]
## Checklist

- [x] The package I am requesting does not already exist on https://chocolatey.org/packages;
- [x] There is no open issue for this package;
- [x] The issue title starts 'RFP - ';
- [x] The download URL is public and not locked behind a paywall / login;

## Package Details

Software project URL : 
http://forums.winamp.com/showthread.php?t=157049

Direct download URL for the software / installer : 
http://stashbox.org/492219/Playlist_File_Remover_v2_5_Beta_1.exe

Software summary / short description: 
Just a little plugin that will allow you to remove a played track from the Winamp playlist automatically. Made by DrO.

**Comment by pauby** [comment]
According to the 'first post in the thread](http://forums.winamp.com/showpost.php?s=330eaa4ccfa5602fa321fa4d06a44e63&p=1172453&postcount=1) that Direct Download URL has been removed (it goes to a 404 page) and the site provided in the footer for the 'stable' version is also a 404 page.

**Comment by EpsilonsQc** [comment]
> According to the [first post in the thread](http://forums.winamp.com/showpost.php?s=330eaa4ccfa5602fa321fa4d06a44e63&p=1172453&postcount=1) that Direct Download URL has been removed (it goes to a 404 page) and the site provided in the footer for the 'stable' version is also a 404 page.

I uploaded the installer to my Google Drive and edited the original post. Hope that help to make the package a reality. This Winamp plugin is so useful.  

Here's the link to the software installer : 
https://drive.google.com/file/d/1QJwv0-1RHcXEicdAHyUJpuV8DJbGjf10

**Comment by pauby** [comment]
@EpsilonsQc We can't use an unofficial download location for software unfortunately. So have it available on your Google Drive is not going to work unfortunately.

**Comment by EpsilonsQc** [comment]
> @EpsilonsQc We can't use an unofficial download location for software unfortunately. So have it available on your Google Drive is not going to work unfortunately.

Well, the situation is kinda sad because this app was made by a random guy on a forum and is nowhere to be found now.

**Comment by pauby** [comment]
Is the source code for it available?

**Comment by EpsilonsQc** [comment]
> Is the source code for it available?

As far as I know, no. This plugin was only available thru the forums post link I shared above.

**Comment by AdmiringWorm** [comment]
@pauby there is possible to traverse the wayback machine to get the original page and the download, not sure if that would be an acceptable solution though.

https://web.archive.org/web/20130210100237/http://nunzioweb.com:80/daz/plfr.html

**Comment by EpsilonsQc** [comment]
> @pauby there is possible to traverse the wayback machine to get the original page and the download, not sure if that would be an acceptable solution though.
> 
> https://web.archive.org/web/20130210100237/http://nunzioweb.com:80/daz/plfr.html

wow that is awesome. the download link work 🐙  !
https://web.archive.org/web/20130210100237/http://nunzioweb.com:80/daz/files/Playlist_File_Remover_v2_2_2.exe

**Comment by AdmiringWorm** [comment]
@EpsilonsQc yes it works, but that link only works in the browser (due to javascript), the direct URL would be: https://web.archive.org/web/20130222021733if_/http://nunzioweb.com/daz/files/Playlist_File_Remover_v2_2_2.exe

**Comment by pauby** [comment]
@AdmiringWorm I didn't even think about that. I would be happy with that if there were comments in the code / description about it. Since it was my initial objection, I'll post in the moderators channel.

**Comment by AdmiringWorm** [comment]
@pauby did we get any final decision regarding using the web archive link?
I honestly don't remember, but have a feeling it leaned more towards not allowing it :confused:

**Comment by pauby** [comment]
@AdmiringWorm We did. But we didn't come back here and I can't remember what the decision was. So I've asked the question again and tagged you in it.

**Comment by pauby** [comment]
@AdmiringWorm Do we have a way forward on this given the chat?

**Comment by AdmiringWorm** [comment]
@pauby yeah, I think we do.

I think it will be safe to open this up to maintainers, with at least one caveat.

For any package for this request, ensure that there is also a `VERIFICATION.txt` file available in the package that details how to validate the binary file.
This is mostly due to the actual webpage only being available on the Wayback machine, as well as the binary file.

I would also like to recommend that anyone that picks up this request, to maybe contact the owner of the website to ensure he/she do not have any objection about the creation of a package for the software.

**Comment by dr0biwan** [comment]
> Well, the situation is kinda sad because this Winamp plugin was made by a random guy on a forum and is nowhere to be found now.

That'd be a random guy who pumped out a load of plug-ins over the years & even worked on Winamp for a while & is easily contactable from the forum profile links (I've only stumbled on this by chance & thought it weird to see one of my plug-ins directly named).

FYI the plug-in was removed as there were issues with it & no the source code hasn't been provided but is instead maintained in an updated form as part of WACUP (my effort to re-create Winamp).

> I would also like to recommend that anyone that picks up this request, to maybe contact the owner of the website to ensure he/she do not have any objection about the creation of a package for the software.

I'm not keen on this happening tbh as I know there's issues with the plug-in (mis-handling of unicode through to live hooking that doesn't work correctly which breaks functionality or can cause stability issues) & a package of it then implies a certain level of support which I'm not offering. But I guess I can't do much about it since the old plug-in files are still out there on the internet.

-dro

**Comment by EpsilonsQc** [comment]
> FYI the plug-in was removed as there were issues with it & no the source code hasn't been provided but is instead maintained in an updated form as part of WACUP (my effort to re-create Winamp).
> 
> I'm not keen on this happening tbh as I know there's issues with the plug-in (mis-handling of unicode through to live hooking that doesn't work correctly which breaks functionality or can cause stability issues) 
> 
> -dro

I'd just like to add that, as someone who still use this plug-in all the time, and has been using it for the past 12+ years, I never encountered any problems/bugs or instability with it. Maybe i'm just lucky. But it pretty much do exactly what it's supposed to do : add an option (hotkey) to remove files from the winamp playlist and place them directly in the recycle bin.
